SourceDistributions/Developer-Sandboxes/tsWxGTUI_PyVx/Python-2x/tsWxGTUI_Py2x/ManPages/tsManPagesLibGUI/tsWxSizerItemList.man

Help on module tsWxSizerItemList:

NAME
    tsWxSizerItemList

FILE
    ./tsLibGUI/tsWxPkg/src/tsWxSizerItemList.py

DESCRIPTION
    tsWxSizerItemList.py - Class wraps a tsWxDoubleLinkedList class
    (rather than a wxList-based class) and gives it a Python
    sequence-like interface. Sequence operations supported are length,
    first, last, next and index access and iteration.

CLASSES
    tsWxDoubleLinkedList.DoubleLinkedList(__builtin__.object)
        SizerItemList
    
    class SizerItemList(tsWxDoubleLinkedList.DoubleLinkedList)
     |  Class wraps a tsWxDoubleLinkedList class (rather than a wxList-based
     |  class) and gives it a Python sequence-like interface. Sequence
     |  operations supported are length, first, last, next and index.
     |  
     |  NOTE: This implementation uses the "object" based DoubleLinkedList
     |  Class. It eliminate the need to build an equivalent object based Class
     |  from scratch.
     |  
     |  Method resolution order:
     |      SizerItemList
     |      tsWxDoubleLinkedList.DoubleLinkedList
     |      __builtin__.object
     |  
     |  Methods defined here:
     |  
     |  __init__(self, userData=None, lifoMode=False)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from tsWxDoubleLinkedList.DoubleLinkedList:
     |  
     |  Add(self, userData)
     |      Add a node to the list tail and appropriately adjust the forward
     |      and backward links.
     |  
     |  FindIndexByNode(self, node)
     |      Return the index associated with the specified node. Return None
     |      if node not found.
     |  
     |  GetCount(self)
     |      Returns the number of children objects managed by the sizer in a
     |      list-type object.
     |      
     |      Modeled after TBD in sizer.cpp file of wxWidgets.
     |  
     |  GetFirst(self)
     |      Returns the first of the children objects managed by the sizer
     |      in a list-type object.
     |      
     |      Modeled after wxSizerItemList::compatibility_iterator in
     |      sizer.cpp file of wxWidgets.
     |  
     |  GetIndex(self, index)
     |      Returns the index specified member of the list.
     |  
     |  GetLast(self)
     |      Returns the last of the children objects managed by the sizer
     |      in a list-type object.
     |      sizer.cpp file of wxWidgets.
     |  
     |  GetNext(self)
     |      Returns the next of the children objects managed by the sizer
     |      in a list-type object.
     |      
     |      Modeled after wxSizerItemList::compatibility_iterator in
     |      sizer.cpp file of wxWidgets.
     |  
     |  InsertAfter(self, afterIndex, userData)
     |      Insert a node to the list after the specified nodetail and
     |      appropriately adjust the forward and backward links.
     |      
     |      function insertAfter(List list, Node node, Node newNode)
     |           newNode.prev := node
     |           newNode.nextNode := node.next
     |           if node.next == null
     |               list.lastNode := newNode
     |           else
     |               node.next.prev := newNode
     |           node.next := newNode
     |  
     |  InsertAsHead(self, userData)
     |      Insert a node to the list before the specified nodetail and
     |      appropriately adjust the forward and backward links.
     |      
     |      function insertBeginning(List list, Node newNode)
     |           if list.firstNode == null
     |               list.firstNode := newNode
     |               list.lastNode  := newNode
     |               newNode.prev := null
     |               newNode.nextNode := null
     |           else
     |               insertBefore(list, list.firstNode, newNode)
     |  
     |  InsertAsTail(self, userData)
     |      Insert a node at the end of the list and appropriately adjust
     |      the forward and backward links.
     |      
     |      function insertEnd(List list, Node newNode)
     |           if list.lastNode == null
     |               insertBeginning(list, newNode)
     |           else
     |               insertAfter(list, list.lastNode, newNode)
     |  
     |  InsertBefore(self, beforeIndex, userData)
     |      Insert a node to the list before the specified nodetail and
     |      appropriately adjust the forward and backward links.
     |      
     |      function insertBefore(List list, Node node, Node newNode)
     |           newNode.prev := node.prev
     |           newNode.nextNode := node
     |           if node.prev == null
     |               list.firstNode := newNode
     |           else
     |               node.prev.next := newNode
     |           node.prev    := newNode
     |  
     |  IsFIFO(self)
     |      Returns True if FIFO (not Lifo) mode.
     |  
     |  IsLIFO(self)
     |      Returns True if LIFO mode.
     |  
     |  Pop(self)
     |      Remove and return node from position appropriate for FIFO (stack)
     |      or LIFO (queue).
     |  
     |  Push(self, userData)
     |      Insert node in position appropriate for FIFO (stack) or LIFO (queue).
     |  
     |  Remove(self, node)
     |      Remove a node from the list and appropriately adjust the forward
     |      and backward links.
     |      
     |      function remove(List list, Node node)
     |         if node.prev == null
     |             list.firstNode := node.next
     |         else
     |             node.prev.next := node.next
     |         if node.next == null
     |             list.lastNode := node.prev
     |         else
     |             node.next.prev := node.prev
     |         destroy node
     |  
     |  __del__(self)
     |  
     |  getNext(self, node)
     |  
     |  getPrevious(self, node)
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from tsWxDoubleLinkedList.DoubleLinkedList:
     |  
     |  Next
     |  
     |  Previous
     |  
     |  __dict__
     |      dictionary for instance variables (if defined)
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)

DATA
    DEBUG = True
    VERBOSE = True
    __authors__ = 'Richard S. Gordon'
    __copyright__ = 'Copyright (c) 2007-2013 Richard S. Gordon.\n\t\tAll r...
    __credits__ = '\n\n  Credits: \n\n\t  tsLibGUI Import & Application .....
    __date__ = '04/01/2013'
    __header__ = '\n\ntsWxSizerItemList, v1.1.0 (build 04/01/2013)\n\n... ...
    __license__ = 'GNU General Public License, Version 3, 29 June 2007'
    __line1__ = 'tsWxSizerItemList, v1.1.0 (build 04/01/2013)'
    __line2__ = 'Author(s): Richard S. Gordon'
    __line3__ = 'Copyright (c) 2007-2013 Richard S. Gordon.\n\t\tAll right...
    __line4__ = 'GNU General Public License, Version 3, 29 June 2...U Gene...
    __title__ = 'tsWxSizerItemList'
    __version__ = '1.1.0'
    mainTitleVersionDate = 'tsWxSizerItemList, v1.1.0 (build 04/01/2013)'

VERSION
    1.1.0

DATE
    04/01/2013

CREDITS
    
    
      Credits: 
    
    	  tsLibGUI Import & Application Launch Features: 
    	  Copyright (c) 2007-2009 Frederick A. Kier.
    			All rights reserved.
    
    	  Python Curses Module API & Run Time Library Features:
    	  Copyright (c) 2001-2013 Python Software Foundation.
    			All rights reserved.
    	  PSF License Agreement for Python 2.7.3 & 3.3.0
    
    	  wxWidgets (formerly wxWindows) & wxPython API Features:
    	  Copyright (c) 1992-2008 Julian Smart, Robert Roebling,
    			Vadim Zeitlin and other members of the 
    			wxWidgets team.
    			All rights reserved.
    	  wxWindows Library License
    
    	  nCurses API & Run Time Library Features:
    	  Copyright (c) 1998-2011 Free Software Foundation, Inc.
    			All rights reserved.
    	  GNU General Public License, Version 3, 29 June 2007




Documents/NEWS.txt
#-----------------------------------------------------------
#"Time-stamp: <07/08/2015  4:48:41 AM rsg>"
#-----------------------------------------------------------

========================= NEWS.txt =========================

   +----+----+  TeamSTARS "tsWxGTUI_PyVx" Toolkit
   | ts | Wx |      with Python 2x & Python 3x based
   +----+----+         Command Line Interface (CLI)
   | G T U I |      and "Curses"-based "wxPython"-style, 
   +---------+         Graphical-Text User Interface (GUI)
 
   Get that cross-platform, pixel-mode "wxPython" feeling
   on character-mode 8-/16-color (xterm-family) & non-color
   (vt100-family) terminals and terminal emulators.

   You can find this and other plain-text files in the
   Toolkit subdirectory named:

       "./<Toolkit Recipient's Repository>/Documents".

       <Your Working Repository>
       (e.g. "tsWxGTUI_PyVx_Repository") 
         |
         |
         +-- ["Documents"]

=========================== News ===========================

The pre-alpha release of the TeamSTARS "tsWxGTUI_PyVx-0.0.0"
Toolkit is now available. It includes the following two
versions abd their associated variants (Python Distribution
Utilities ("Distutils") installable "Site-Packages" and non-
installable "Developer-Sandboxes" for enhancement and bug
fix development and experimentation without interference
with the already installed Site-Packages):

1. The TeamSTARS "tsWxGTUI_Py2x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   second generation Python 2.x programming language which
   is widely available.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. The final 2.x
   version 2.7 release came out in mid-2010, with a state-
   ment of extended support for this end-of-life release.
   The 2.x branch will see no new major releases after that.

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 2.0.0-2.7.9.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 2.6.4-2.7.9.

2. The TeamSTARS "tsWxGTUI_Py3x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   third generation Python 3.x programming language which
   is growing in availability.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. 3.x is under
   active development and has already seen over five years
   of stable releases, including version 3.3 in 2012 and
   3.4 in 2014. This means that all recent standard library
   improvements, for example, are only available by default
   in Python 3.x.

   Guido van Rossum (the original creator of the Python
   language) decided to clean up Python 2.x properly,
   with less regard for backwards compatibility than is
   the case for new releases in the 2.x range. The most
   drastic improvement is the better Unicode support
   (with all text strings being Unicode by default) as
   well as saner bytes/Unicode separation.

   Besides, several aspects of the core language (such as
   print and exec being statements, integers using floor
   division) have been adjusted to be easier for newcomers
   to learn and to be more consistent with the rest of the
   language, and old cruft has been removed (for example,
   all classes are now new-style, "range()" returns a memory
   efficient iterable, not a list as in 2.x).

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 3.0.0-3.4.2.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 3.1.5-3.4.2.

======================= End-Of-File ========================


SourceDistributions/Developer-Sandboxes/tsWxGTUI_PyVx/Python-3x/tsWxGTUI_Py3x/tsLibGUI/tsWxPkg/src/tsWxSizerItemList.py
#! /usr/bin/env python
# "Time-stamp: <10/25/2013  7:35:41 AM rsg>"
'''
tsWxSizerItemList.py - Class wraps a tsWxDoubleLinkedList class
(rather than a wxList-based class) and gives it a Python
sequence-like interface. Sequence operations supported are length,
first, last, next and index access and iteration.
'''
#################################################################
#
# File: tsWxSizerItemList.py
#
# Purpose:
#
#    Class wraps a tsWxDoubleLinkedList class (rather than a
#    wxList-based class) and gives it a Python sequence-like
#    interface. Sequence operations supported are length,
#    first, last, next and index.
#
# Usage (example):
#
#    # Import
#
#    from tsWxSizerItemList import SizerItemList
#
# Requirements:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Capabilities:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Limitations:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Notes:
#
#    None
#
# Methods:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Classes:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Modifications:
#
#    2012/06/02 rsg Changed InsertBeginning to InsertAsHead. Changed
#                   InsertEnd to InsertAsTail.
#
#    2012/06/02 rsg Added support for optional LIFO (Last In First Out)
#                   "stack" mode or default FIFO (First In First Out)
#                   "queue" mode to DoubleLinkedList.
#
#    2012/06/02 rsg Updated test cases to replicate the lates ones
#                   used for base class tsWxDoubleLinkedList.
#
# ToDo:
#
#    None.
#
#################################################################

__title__     = 'tsWxSizerItemList'
__version__   = '1.1.0'
__date__      = '04/01/2013'
__authors__   = 'Richard S. Gordon'
__copyright__ = 'Copyright (c) 2007-2013 ' + \
                '%s.\n\t\tAll rights reserved.' % __authors__
__license__   = 'GNU General Public License, ' + \
                'Version 3, 29 June 2007'
__credits__   = '\n\n  Credits: ' + \
                '\n\n\t  tsLibGUI Import & Application Launch Features: ' + \
                '\n\t  Copyright (c) 2007-2009 Frederick A. Kier.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\n\t  Python Curses Module API & ' + \
                'Run Time Library Features:' + \
                '\n\t  Copyright (c) 2001-2013 ' +\
                'Python Software Foundation.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  PSF License Agreement for Python 2.7.3 & 3.3.0' + \
                '\n\n\t  wxWidgets (formerly wxWindows) & ' + \
                'wxPython API Features:' + \
                '\n\t  Copyright (c) 1992-2008 Julian Smart, ' + \
                'Robert Roebling,' + \
                '\n\t\t\tVadim Zeitlin and other members of the ' + \
                '\n\t\t\twxWidgets team.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  wxWindows Library License' + \
                '\n\n\t  nCurses API & Run Time Library Features:' + \
                '\n\t  Copyright (c) 1998-2011 ' + \
                'Free Software Foundation, Inc.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  GNU General Public License, ' + \
                'Version 3, 29 June 2007'

__line1__ = '%s, v%s (build %s)' % (__title__, __version__, __date__)
__line2__ = 'Author(s): %s' % __authors__
__line3__ = '%s' % __copyright__

if len( __credits__) == 0:
    __line4__ = '%s' % __license__
else:
    __line4__ = '%s%s' % (__license__, __credits__)

__header__ = '\n\n%s\n\n  %s\n  %s\n  %s\n' % (__line1__,
                                               __line2__,
                                               __line3__,
                                               __line4__)

mainTitleVersionDate = __line1__

from tsWxDoubleLinkedList import DoubleLinkedList

##import tsWxGlobals as wx # Invalid import because of circular dependency.
import tsCxGlobals as cx

#---------------------------------------------------------------------------

##DEBUG = True # Avoids tsWx import because of circular dependency.
DEBUG = cx.ThemeToUse['Troubleshooting']['Debug_GUI_Launch'] | \
        cx.ThemeToUse['Troubleshooting']['Debug_GUI_Progress'] | \
        cx.ThemeToUse['Troubleshooting']['Debug_GUI_Termination'] | \
        cx.ThemeToUse['Troubleshooting']['Debug_GUI_Exceptions']

##VERBOSE = True # Avoids tsWx import because of circular dependency.
VERBOSE = cx.ThemeToUse['Troubleshooting']['Debug_GUI_Configuration']

#---------------------------------------------------------------------------
 
class SizerItemList(DoubleLinkedList):
    '''
    Class wraps a tsWxDoubleLinkedList class (rather than a wxList-based
    class) and gives it a Python sequence-like interface. Sequence
    operations supported are length, first, last, next and index.

    NOTE: This implementation uses the "object" based DoubleLinkedList
    Class. It eliminate the need to build an equivalent object based Class
    from scratch.
    '''
    def __init__(self, userData=None, lifoMode=False):
        '''
        '''
##        theClass = 'SizerItemList'

##        wx.RegisterFirstCallerClassName(self, theClass)

##        self.tsBeginClassRegistration(theClass, id)

        DoubleLinkedList.__init__(self, userData=userData, lifoMode=lifoMode)

##        self.tsEndClassRegistration(theClass)

    #-----------------------------------------------------------------------
    # Begin tsWx API Extensions

    # End tsWx API Extensions
    #-----------------------------------------------------------------------

#---------------------------------------------------------------------------

if __name__ == '__main__':

    #-----------------------------------------------------------------------

    def test_low_level_interface():
        '''
        '''

        #-------------------------------------------------------------------

        myList = DoubleLinkedList("a")
        myList.Add("b")
        myList.Add("c")
        myList.Add("d")
        myList.Add("e")
        firstNode = myList.GetFirst()
        lastNode = myList.GetLast()

        #-------------------------------------------------------------------

        node = firstNode
        done = False
        print('\n***** Five Member Set [abcde] after ' + \
              'low level next/previous access')
        while not done:
            if node.previous is None:
                textPrevious = None
            else:
                textPrevious = node.previous.userData

            textCurrent = node.userData

            if node.__next__ is None:
                textNext = None
            else:
                textNext = node.next.userData

            print('\tPrevious="%s"; userData="%s"; Next="%s"' % (
                textPrevious, textCurrent, textNext))

            if node.__next__ is None:
                done = True
                print('\n')
            else:
                node = node.__next__

        #-------------------------------------------------------------------

        node = firstNode
        done = False
        print('\n***** Five Member Set [abcde] after ' + \
              'low level forward access')
        while not done:
            print('\t"%s"' % node.userData)
            if node.__next__ is None:
                done = True
                print('\n')
            else:
                node = node.__next__

        #-------------------------------------------------------------------

        node = lastNode
        done = False
        print('\n***** Five Member Set [edcba] after ' + \
              'low level backward access')
        while not done:
            print('\t"%s"' % node.userData)
            if node.previous is None:
                done = True
                print('\n')
            else:
                node = node.previous

        #-------------------------------------------------------------------

        myList.Remove(myList.tail)
        node = myList.head
        done = False
        print('\n***** Four Member Set [abcd] after ' + \
              'low level tail removal')
        while not done:
            print('\t"%s"' % node.userData)
            if node.__next__ is None:
                done = True
                print('\n')
            else:
                node = node.__next__

        #-------------------------------------------------------------------

        myList.Remove(myList.head)
        node = myList.head
        done = False
        print('\n***** Three Member Set [bcd] after ' + \
              'low level head removal')
        while not done:
            print('\t"%s"' % node.userData)
            if node.__next__ is None:
                done = True
                print('\n')
            else:
                node = node.__next__

        #-------------------------------------------------------------------

        myList.Remove(myList.head.__next__)
        node = myList.head
        done = False
        print('\n***** Two Member Set (bd) after ' + \
              'low level internal removal')
        while not done:
            print('\t"%s"' % node.userData)
            if node.__next__ is None:
                done = True
                print('\n')
            else:
                node = node.__next__

   #-----------------------------------------------------------------------

    def test_high_level_interface():
        '''
        '''

        #------------------------------------------------------------------

        myList = DoubleLinkedList("a")
        myList.Add("b")
        myList.Add("c")
        myList.Add("d")
        myList.Add("e")

        print('\n***** Five Member Set [abcde] with ' + \
                       'high level iterator access')
        theCount = myList.GetCount()
        print('\ttheCount="%s"' % theCount)

        index = 0
        node = myList.GetFirst()
        while (not (node is None)):
            theGetUserData = node.GetUserData()
            print('\tnode=[%d]; userData="%s"; theCount=%d' % (
                index,
                theGetUserData,
                myList.GetCount()))
            index += 1
            node = myList.GetNext()

        #------------------------------------------------------------------

        myList = DoubleLinkedList("a")
        myList.Add("b")
        myList.Add("c")
        myList.Add("d")
        myList.Add("e")

        print('\n***** Five Member Set {??abcde??] with high level ' + \
              'forward index access')
        theCount = myList.GetCount()
        print('\ttheCount="%s"' % theCount)
        for index in range(0 - 2, theCount + 2):

            node = myList.GetIndex(index)
            if node is None:
                theGetUserData = '?'
            else:
                theGetUserData = node.GetUserData()
            print('\tnode=[%d]; userData="%s"; theCount=%d' % (
                index,
                theGetUserData,
                myList.GetCount()))

        #------------------------------------------------------------------

        myList = DoubleLinkedList("a")
        myList.Add("b")
        myList.Add("c")
        myList.Add("d")
        myList.Add("e")

        print('\n***** Five Member Set {??edcba??] with high level ' + \
              'reverse index access')
        theCount = myList.GetCount()
        print('\t     theCount="%s"' % theCount)
        for index in range(theCount + 1, 0 - 3, -1):

            node = myList.GetIndex(index)
            if node is None:
                theGetUserData = '?'
            else:
                theGetUserData = node.GetUserData()
            print('\tnode=[%d]; userData="%s"; theCount=%d' % (
                index,
                theGetUserData,
                myList.GetCount()))

        #------------------------------------------------------------------

        myList = DoubleLinkedList()
        myList.InsertAsTail("a")
        myList.InsertAsTail("b")
        myList.InsertAsTail("c")
        myList.InsertAsTail("d")
        myList.InsertAsTail("e")

        print(
            '\n***** Five Member Set [abcde] with high level InsertAsTail access')
        theCount = myList.GetCount()
        print('\ttheCount="%s"' % theCount)
        index = 0
        node = myList.GetFirst()
        while (not (node is None)):
            theGetUserData = node.GetUserData()
            print('\tnode=[%d]; userData="%s"; theCount=%d' % (
                index,
                theGetUserData,
                myList.GetCount()))
            index += 1
            node = myList.GetNext()

        #------------------------------------------------------------------

        myList = DoubleLinkedList()
        myList.InsertAsHead("a")
        myList.InsertAsHead("b")
        myList.InsertAsHead("c")
        myList.InsertAsHead("d")
        myList.InsertAsHead("e")

        print('\n***** Five Member Set [edcba] with high level ' + \
              'InsertAsHead access')
        theCount = myList.GetCount()
        print('\ttheCount="%s"' % theCount)
        index = 0
        node = myList.GetFirst()
        while (not (node is None)):
            theGetUserData = node.GetUserData()
            print('\tnode=[%d]; userData="%s"; theCount=%d' % (
                index,
                theGetUserData,
                myList.GetCount()))
            index += 1
            node = myList.GetNext()

        #------------------------------------------------------------------

        myList = DoubleLinkedList()
        myList.InsertAsHead("a")
        myList.InsertBefore(0, "b")
        myList.InsertBefore(0, "c")
        myList.InsertBefore(0, "d")
        myList.InsertBefore(0, "e")

        print('\n***** Five Member Set [edcba] with high level ' + \
              'InsertBefore access')
        theCount = myList.GetCount()
        print('\ttheCount="%s"' % theCount)
        index = 0
        node = myList.GetFirst()
        while (not (node is None)):
            theGetUserData = node.GetUserData()
            print('\tnode=[%d]; userData="%s"; theCount=%d' % (
                index,
                theGetUserData,
                myList.GetCount()))
            index += 1
            node = myList.GetNext()

        #------------------------------------------------------------------

        myList = DoubleLinkedList()
        myList.InsertAsHead("a")
        myList.InsertAfter(0, "b")
        myList.InsertAfter(1, "c")
        myList.InsertAfter(0, "d")
        myList.InsertAfter(2, "e")

        print('\n***** Five Member Set [adbec] with high level ' + \
              'InsertAfter access')
        theCount = myList.GetCount()
        print('\ttheCount="%s"' % theCount)
        index = 0
        node = myList.GetFirst()
        while (not (node is None)):
            theGetUserData = node.GetUserData()
            print('\tnode=[%d]; userData="%s"; theCount=%d' % (
                index,
                theGetUserData,
                myList.GetCount()))
            index += 1
            node = myList.GetNext()

        #------------------------------------------------------------------

        myList = DoubleLinkedList(lifoMode=False)
        myList.Push("a")
        myList.Push("b")
        myList.Push("c")
        myList.Push("d")
        myList.Push("e")

        print('\n***** Five Member FIFO Set [abcde] with ' + \
                       'high level Push and Pop access')
        theCount = myList.GetCount()
        print('\ttheCount="%s"' % theCount)

        index = 0
        node = myList.Pop()
        while (not (node is None)):
            theGetUserData = node.GetUserData()
            print('\tnode=[%d]; userData="%s"; theCount=%d' % (
                index,
                theGetUserData,
                myList.GetCount()))
            index += 1
            node = myList.Pop()

        #------------------------------------------------------------------

        myList = DoubleLinkedList(lifoMode=True)
        myList.Push("a")
        myList.Push("b")
        myList.Push("c")
        myList.Push("d")
        myList.Push("e")

        print('\n***** Five Member LIFO Set [edcba] with ' + \
                       'high level Push and Pop access')
        theCount = myList.GetCount()
        print('\ttheCount="%s"' % theCount)

        index = 0
        node = myList.Pop()
        while (not (node is None)):
            theGetUserData = node.GetUserData()
            print('\tnode=[%d]; userData="%s"; theCount=%d' % (
                index,
                theGetUserData,
                myList.GetCount()))
            index += 1
            node = myList.Pop()

   #-----------------------------------------------------------------------

    print(__header__)

    test_low_level_interface()
    test_high_level_interface()
